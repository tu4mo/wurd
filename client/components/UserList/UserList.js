import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import FollowButton from '../FollowButton'
import { Table, Tr, Td } from '../Table'
import ProfilePhoto from '../ProfilePhoto'
import Spinner from '../Spinner'
import './UserList.scss'

const UserList = ({ isPending, users }) => {
  if (isPending) {
    return <Spinner />
  }

  if (!users || users.length === 0) {
    return null
  }

  return (
    <Table className="user-list">
      {users.map(({ username, profileUrl }) =>
        <Tr key={username}>
          <Td>
            <div className="user-list__user">
              <Link to={`/${username}`}>
                <ProfilePhoto
                  size="small"
                  url={profileUrl}
                  username={username}
                />
              </Link>
              <Link className="user-list__username" to={`/${username}`}>
                {username}
              </Link>
            </div>
          </Td>
          <Td style={{ textAlign: 'right', width: '1%' }}>
            <FollowButton username={username} />
          </Td>
        </Tr>
      )}
    </Table>
  )
}

UserList.propTypes = {
  isPending: PropTypes.bool,
  users: PropTypes.array
}

export default UserList
