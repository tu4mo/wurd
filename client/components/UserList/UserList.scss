.user-list {
  margin: 0 auto;
  max-width: 640px;

  &__username {
    margin-left: var(--spacing-sm);
    text-decoration: none;
  }
}
