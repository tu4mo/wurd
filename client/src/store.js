import { createStore, applyMiddleware, compose } from 'redux'
import thunk from 'redux-thunk'
import reducers from './reducers'
import api from './api'

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
const persistedState = localStorage.getItem('auth')
  ? { auth: JSON.parse(localStorage.getItem('auth')) }
  : {}

const store = createStore(
  reducers,
  persistedState,
  composeEnhancers(applyMiddleware(thunk.withExtraArgument(api)))
)

store.subscribe(() => {
  localStorage.setItem('auth', JSON.stringify(store.getState().auth))
})

export default store
