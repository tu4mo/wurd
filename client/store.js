import { createStore, applyMiddleware, compose } from 'redux'
import thunk from 'redux-thunk'
import reducers from './reducers'
import api from './api'

const composeEnhancers =
  process.env.NODE_ENV === 'production' ||
  process.env.NODE_ENV === 'test' ||
  !window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? compose
    : window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__

const store = createStore(
  reducers,
  composeEnhancers(applyMiddleware(thunk.withExtraArgument(api)))
)

export default store
