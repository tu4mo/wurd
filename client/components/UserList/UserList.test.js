import React from 'react'
import { MemoryRouter } from 'react-router'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import store from '../../store'

import UserList from './UserList'

it('renders correctly', () => {
  const tree1 = renderer.create(<UserList />).toJSON()

  const tree2 = renderer.create(<UserList isPending />).toJSON()

  const tree3 = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <UserList
            users={[
              {
                profileUrl: 'https://www.domain.com/profile.jpg',
                username: 'tu4mo'
              }
            ]}
          />
        </MemoryRouter>
      </Provider>
    )
    .toJSON()

  expect(tree1).toMatchSnapshot()
  expect(tree2).toMatchSnapshot()
  expect(tree3).toMatchSnapshot()
})
