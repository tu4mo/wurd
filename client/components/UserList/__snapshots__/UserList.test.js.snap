// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `null`;

exports[`renders correctly 2`] = `
<div
  className="spinner"
/>
`;

exports[`renders correctly 3`] = `
<div
  className="user-list sc-bdVaJa kEgatI"
>
  <div
    className="sc-bwzfXH ekItIQ"
  >
    <div
      className="user-list__row"
    >
      <div
        className="user-list__user"
      >
        <a
          href="/tu4mo"
          onClick={[Function]}
        >
          <div
            className="sc-htpNat gmRwLL"
          >
            <img
              alt="tu4mo"
              className="image"
              src="https://www.domain.com/profile.jpg"
              title="tu4mo"
            />
          </div>
        </a>
        <a
          className="user-list__username"
          href="/tu4mo"
          onClick={[Function]}
        >
          tu4mo
        </a>
      </div>
    </div>
  </div>
</div>
`;
